Geometry_Engine/Modify/RoundCoordinates.cs

```diff
@@ -0,0 +1,309 @@
+        [Input("polyline", "The BHoM Geometry Polyline to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry Polyline.")]
+        public static Polyline RoundCoordinates(this Polyline polyline, int decimalPlaces = 6)
+        {
+            return new Polyline()
+            {
+                ControlPoints = polyline.ControlPoints.Select(x => x.RoundCoordinates(decimalPlaces)).ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry PolyCurve's curves to be rounded to the number of provided decimal places.")]
+        [Input("polycurve", "The BHoM Geometry PolyCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry PolyCurve.")]
+        public static PolyCurve RoundCoordinates(this PolyCurve polycurve, int decimalPlaces = 6)
+        {
+            return new PolyCurve()
+            {
+                Curves = polycurve.Curves.Select(x => x.IRoundCoordinates(decimalPlaces)).ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry NurbsCurve's control points to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry NurbsCurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM Geometry NurbsCurve.")]
+        public static NurbsCurve RoundCoordinates(this NurbsCurve nurbscurve, int decimalPlaces = 6)
+        {
+            return new NurbsCurve()
+            {
+                ControlPoints = nurbscurve.ControlPoints.Select(x => x.RoundCoordinates(decimalPlaces)).ToList(),
+                Knots = nurbscurve.Knots.ToList(),
+                Weights = nurbscurve.Weights.ToList()
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry PlanarSurface's defining curves to be rounded to the number of provided decimal places.")]
+        [Input("nurbscurve", "The BHoM Geometry PlanarSurface to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("surface", "The modified BHoM Geometry PlanarSurface.")]
+        public static PlanarSurface RoundCoordinates(this PlanarSurface planarSurface, int decimalPlaces = 6)
+        {
+            ICurve externalBoundery = planarSurface.ExternalBoundary.IRoundCoordinates(decimalPlaces);
+            List<ICurve> internalBounderies = planarSurface.InternalBoundaries.Select(x => x.IRoundCoordinates(decimalPlaces)).ToList();
```

Comment:
Besides that, this is likely to invalidate the planarity requirement set to `PlanarSurface`. Thinking about it now, I cannot imagine how this could work - I would rather tend to record a warning that would explain the problem and recommend extracting plane => rounding it => projecting the surface on rounded plan as a semi-manual workaround. Return item could simply be the input (or null, not sure tbh).

Plus, boundAry

Reply:
> extracting plane => rounding it => projecting the surface on rounded plan as a semi-manual workaround.

Don't think that would round the coordinates since every point on a rounded Plane isn't a rounded point.

And I was thinking that for:

- 90% would be in an orthogonal plane (XY, YZ, XZ), and hence work,
- 9% would be vertical quads, and hence work,
- the rest would fail but didn't really know of a way to fix that

Thinking about it some more I think one could essentially transform the plane down to the XY-plane, round there and reverse.
Would be kind of finicky to get the correct transformation where an arbitrary planes would intersect the rounded coordinates to the rounded coordinates on the XY plane.

But think that kind of functionality can wait. (and would need to be discussed around how it would act)

Reply:
Hmm, so maybe a solution half way:
- if normal of the surface is parallel to one of the global axes, then round its plane and then round the outline curves
- else record a warning and recommend a semi-manual solution

Could be an easy win for most cases while being safe for others?